core/src/trezor/lvglui/scrs/fingerprints.py

```
@@ -258,60 +448,101 @@
+                break
+            enroll_task = request_enroll(register_count % FP_TEMPLATE_ENROLL_COUNT)
+            cancel_task = progress.request()
+            racer = loop.race(enroll_task, cancel_task)
+            await racer
+            if cancel_task in racer.finished:
+                return False
+            register_count += 1
+            if (register_count) % FP_TEMPLATE_ENROLL_COUNT == 0:
+                if not fingerprint.register_template(
+                    available_positions[register_count // FP_TEMPLATE_ENROLL_COUNT - 1]
+                ):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)